--- Day 1: Report Repair ---

After saving Christmas five years in a row, you've decided to take a vacation at a nice resort on a tropical island. Surely, Christmas will go on without you.

The tropical island has its own currency and is entirely cash-only. The gold coins used there have a little picture of a starfish; the locals just call them stars. None of the currency exchanges seem to have heard of them, but somehow, you'll need to find fifty of these coins by the time you arrive so you can pay the deposit on your room.

To save your vacation, you need to get all fifty stars by December 25th.

Collect stars by solving puzzles. Two puzzles will be made available on each day in the Advent calendar; the second puzzle is unlocked when you complete the first. Each puzzle grants one star. Good luck!

Before you leave, the Elves in accounting just need you to fix your expense report (your puzzle input); apparently, something isn't quite adding up.

Specifically, they need you to find the two entries that sum to 2020 and then multiply those two numbers together.

For example, suppose your expense report contained the following:

1721    
979 
366 
299 
675 
1456    

In this list, the two entries that sum to 2020 are 1721 and 299. Multiplying them together produces 1721 * 299 = 514579, so the correct answer is 514579.

Of course, your expense report is much larger. Find the two entries that sum to 2020; what do you get if you multiply them together?


In [1]:
with open('Input1.txt','r') as f:
    read_data = f.read().split('\n')

read_data = [int(x) for x in read_data]

for i in read_data:
    for second in read_data:
        if i + second == 2020:
            print(i, second)
            print(i*second)
            break

1958 62
121396
62 1958
121396


Your puzzle answer was 121396.

The first half of this puzzle is complete! It provides one gold star: *

--- Part Two ---

The Elves in accounting are thankful for your help; one of them even offers you a starfish coin they had left over from a past vacation. They offer you a second one if you can find three numbers in your expense report that meet the same criteria.

Using the above example again, the three entries that sum to 2020 are 979, 366, and 675. Multiplying them together produces the answer, 241861950.

In your expense report, what is the product of the three entries that sum to 2020?


In [2]:
for i in read_data:
    for second in read_data:
        for third in read_data:
            if i + second + third == 2020:
                print(i, second, third)
                print(i*second*third)
                break

426 1477 117
73616634
426 117 1477
73616634
1477 426 117
73616634
1477 117 426
73616634
117 426 1477
73616634
117 1477 426
73616634
